[RU] Add translation for Guardian lesson

brain-geek opened the pull request.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+lang: ru
+---
+
+[Guardian](https://github.com/ueberauth/guardian) - часто используемая библиотека, основанная на [JWT](https://jwt.io/) (Javascript Web Token).
```
`-` → `&mdash;`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Что такое JWT? JWT занимается созданием токена для аутентификации. В то время как многие системы аутентификации предоставляют доступ только к идентификатору ресурса, JWT позволяет хранить еще и другую информацию, такую как:
+
+* Кто выдал токен
```
In the following list items the verb `выдать` is used in the passive voice so I'd suggest to replace `Кто выдал` with `Кем выдан` for consistency.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+* Каков тип токена
+* Какие права есть у владельца данного токена
+
+Это всё - простые примеры полей в JWT. Технология позволяет добавлять любую дополнительную информацию, которой требует логика приложения. Главное - не забывайте что этот токен должен быть коротким, ведь он должен поместиться в HTTP заголовке.
```
`-` → `&mdash;` in both places.

I also thought that we probably could mention the suggested maximum HTTP header length. There is no limit in the [specification](https://tools.ietf.org/html/rfc7230#section-3.1.1) but most of the servers use the default maximum value (~8KB).
/cc @doomspork

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+Токены JWT могут использваться для аутентификации в любой части приложения:
+
+* Одностраничные приложения
+* Контроллеры (как с помощью браузерной сессии, так и api: с помощью http заголовков)
```
I think we should use capitalized `HTTP` in the whole document.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+### Нужно ли использовать базу данных?
+
+Нет необходимости в отслеживания JWT токенов с использованиием базы данных. Достаточно основываться на том факте, что они уже выписаны и использовать время жизни токена для контроля доступа. Часто встречаются случаи, когда нужно сделать запрос в базу для поиска/получения нужного ресурса, однако JWT это не требует.
```
`Нет необходимости в отслеживания JWT токенов с использованиием базы данных.` → `JWT токены не нужно отслеживать с помощью базы данных.`
`Достаточно основываться на том факте, что` → `Достаточно знать, что`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Нет необходимости в отслеживания JWT токенов с использованиием базы данных. Достаточно основываться на том факте, что они уже выписаны и использовать время жизни токена для контроля доступа. Часто встречаются случаи, когда нужно сделать запрос в базу для поиска/получения нужного ресурса, однако JWT это не требует.
+
+Например, если нужно аутентифицировать подключение к UDP сокету с помощью JWT, подключение к базе данных скорее всего не понадобится. Вся необходимая информация уже будет встроена в токен в момент его создания. Как только подпись проверена - можно использовать данные из токена.
```
`Как только подпись проверена - можно использовать данные из токена.` → `Данные из токена можно использовать сразу после проверки подписи.`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Например, если нужно аутентифицировать подключение к UDP сокету с помощью JWT, подключение к базе данных скорее всего не понадобится. Вся необходимая информация уже будет встроена в токен в момент его создания. Как только подпись проверена - можно использовать данные из токена.
+
+Однако, _можно_ использовать базу данных для отслеживания JWT токенов. Если воспользоваться этой возможностью, можно проверить что токен не был отозван. Как вариант - можно использовать записи в базе данных для отзыва сразу всех токенов пользователя. Это довольно просто реализуется в Guardian с помощью [GuardianDb](https://github.com/hassox/guardian_db). GuardianDb использует интерфейсы Guardian  ...
```
` - ` → `, `

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+  serializer: MyApp.GuardianSerializer
+```
+
+Это минимальный набор информации, который нужно предоставить Guardian для функционирования. Не стоит встраивать секретный ключ напрямую в основную конфигурацию приложения. Вместо этого - у каждого окружения должен быть свой ключ. Хорошей практикой является использовать название Mix окружения в качестве ключа для dev и test окружений. В то же время, staging и production окружения должны использова ...
````
` - ` is not needed here.

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Сериализатор отвечает за поиск нужного ресурса, идентифицируемого по полю сериализованным данным внутри токена. Это может быть поиск в базе данных, внешнем API - или даже простая строка.
````
`сериализованных данных`? Probably the first sentence could be rephrased. I was unable to understand it until I took a peek at the original one.
Also there is another unnecessary `-`.

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+```
+
+Сериализатор отвечает за поиск нужного ресурса, идентифицируемого по полю сериализованным данным внутри токена. Это может быть поиск в базе данных, внешнем API - или даже простая строка.
+Он также отвечает за обратный процесс - превращения объекта пользователя в сериализованный вариант.
````
And this `-` is not needed if you use `превращения`. An `&mdash;` would be needed if you had used `превращение` instead.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+## HTTP запросы
+
+Guardian предоставляет различные Plug для интеграции в HTTP запросы. О Plug можно почитать в [другом уроке](../specifics/plug/). Guardian не требует Phoenix, но при c tuj использованием легче всего будет привести следующие примеры.
```
There is something wrong with `но при c tuj использованием`.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Guardian предоставляет различные Plug для интеграции в HTTP запросы. О Plug можно почитать в [другом уроке](../specifics/plug/). Guardian не требует Phoenix, но при c tuj использованием легче всего будет привести следующие примеры.
+
+Самый легкий способ интеграции - через роутер. Так как HTTP интеграции Guardian основаны на plug-ах, их можно использовать в любом совместимом инструментарии.
```
` - через роутер` → `&mdash; роутер`.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+1. Найти токен внутри запроса и проверить его: `Verify...` plug-и.
+2. Опционально загрузить ресурсы, которые указаны в токене: plug `LoadResource`.
+3. Проверить, что найден пользователь. И если нет - запретить доступ: plug `EnsureAuthenticated`.
```
`И если нет - ` → `Иначе`

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Эти pipeline могут быть использованы для того, чтобы собрать из них реализацию различных требований к аутентификации. Первый вариант - пытается найти токен сначала в сессии, а потом - в заголовках запроса, и если находит - загружает ресурс.
````
`Первый вариант сначала пытается найти токен  в сессии, потом &mdash; в заголовках запроса, и, если находит, загружает ресурс.`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Эти pipeline могут быть использованы для того, чтобы собрать из них реализацию различных требований к аутентификации. Первый вариант - пытается найти токен сначала в сессии, а потом - в заголовках запроса, и если находит - загружает ресурс.
+
+Второй - требует наличия корректного проверенного токена и то, что его тип `access`. Для использования, их нужно добавить в роутер:
```
```
Второй &mdash; требует наличия корректного проверенного токена типa `access`.
```
`, их` → `их`

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Путь `login` в примере выше будет с аутентифицированными пользователем (если таковой есть). Во втором же примере мы удостоверяемся что токен присутствует и правилен. Нет никаких требований по обязательному наличию их в процессе обработки запросов, можно добавить их только в тех местах, где это необходимо.
````
`удостоверяемся что` → `проверяем, что`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Путь `login` в примере выше будет с аутентифицированными пользователем (если таковой есть). Во втором же примере мы удостоверяемся что токен присутствует и правилен. Нет никаких требований по обязательному наличию их в процессе обработки запросов, можно добавить их только в тех местах, где это необходимо.
+
+На данный момент всё ещё не хватает одной части - обработчика ошибок, которые возникают в `EnsureAuthenticated`. Это очень простой модуль, у которого должно быть реализовано две функции:
```
`-` → `&mdash;`
`, которые возникают` → `возникающих`
`должно быть реализовано` → `должны быть реализованы`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+#### Контроллер
+
+В контроллере есть несколько вариантов как получить доступ к текущему пользователю. Начнем с самого простого:
```
`несколько вариантов как получить доступ` → `несколько вариантов получения доступа`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+После подключения модуля `Guardian.Phoenix.Controller` обработчики будут получать два дополнительных параметра, по которым можно делать сопоставление с образцом. Стоит заметить, что если не использовался `EnsureAuthenticated`, то вместо пользователя может быть nil.
+
+Второй - более гибкий/очевидный вариант - использовать специальные функции самого `Guardian`.
```
`Второй, более гибкий/очевидный вариант &mdash; использовать`

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Когда речь идет о авторизации в API &mdash; ситуация слегка отличается, ведь тогда нет сессии и нужно передать токен клиенту.
````
`о авторизации` → `об авторизации`

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Реализация логина через браузерную сессию вызывает `encode_and_sign` самостоятельно, внутри имплементации `Guardian.Plug.sign_in`. Потому оба приведеных кода эквивалентны.
````
```
самостоятельно вызывает `encode_and_sign` внутри имплементации `Guardian.Plug.sign_in`
```
`приведеных` → `приведёных`
I think `оба кода` is not a correct way to say this. How about `листинга`, `примера` or `варианта`?

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+  plug Guardian.Plug.LoadResource
+end
+
+pipeline :ensure_authed_acces do
```
`:ensure_authed_acces` → `:ensure_authed_access`

brain-geek commented:
@nscyclone , thanks for looking through! 

Can you check once again?

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Это всё &mdash; простые примеры полей в JWT. Технология позволяет добавлять любую дополнительную информацию, которой требует логика приложения. Главное &mdash; не забывайте что этот токен должен быть коротким, ведь он должен поместиться в HTTP заголовке.
+
+Следствием такого подхода является тот факт, что JWT могут передаваться как полноценный целостный элемент данных идентификации пользователя.
```
`Следствием такого подхода является тот факт, что JWT могут передаваться`  sounds complicated. I'd suggest using `В результате такого подхода JWT могут передаваться`.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+### Когда их использовать
+
+Токены JWT могут использваться для аутентификации в любой части приложения:
```
`использваться` → `использоваться`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+* Phoenix Channels
+* Запросы между сервисами
+* Запросы между процессами
+* Функциональность "запомнить меня"
```
In the original lesson `3rd Party access (OAuth)` and `Other interfaces - raw TCP, UDP, CLI` are also mentioned in this list.

brain-geek replied:
Yes, it translated very badly and I ommited one or two points.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+### Нужно ли использовать базу данных?
+
+JWT токены не нужно отслеживать с помощью базы данных. Достаточно знать, что
+ они уже выписаны и использовать время жизни токена для контроля доступа. Часто встречаются случаи, когда нужно сделать запрос в базу для поиска/получения нужного ресурса, однако JWT это не требует.
```
Probably this line break was made by mistake.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+
+Например, если нужно аутентифицировать подключение к UDP сокету с помощью JWT, подключение к базе данных скорее всего не понадобится. Вся необходимая информация уже будет встроена в токен в момент его создания. Данные из токена можно использовать сразу после проверки подписи.
+
+Однако, _можно_ использовать базу данных для отслеживания JWT токенов. Если воспользоваться этой возможностью, можно проверить что токен не был отозван. Как вариант, можно использовать записи в базе данных для отзыва сразу всех токенов пользователя. Это довольно просто реализуется в Guardian с помощью [GuardianDb](https://github.com/hassox/guardian_db). GuardianDb использует интерфейсы Guardian д ...
```
`можно проверить что` → `можно проверить, что`

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+2. Опционально загрузить ресурсы, которые указаны в токене: plug `LoadResource`.
+3. Проверить, что найден пользователь. Иначе &mdash; запретить доступ: plug `EnsureAuthenticated`.
+
+Для большей гибкости, Guardian реализует все эти фазы отдельными plug-ами. 
```
`гибкости, Guardian` → `гибкости Guardian`

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Эти pipeline могут быть использованы для того, чтобы собрать из них реализацию различных требований к аутентификации. Первый вариант сначала пытается найти токен в сессии, потом &mdash; в заголовках запроса, и, если находит, загружает ресурс.
````
`для того` can be safely removed here.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+  use Guardian.Phoenix.Controller
+  
+  def some_action(conn, params, user, claims) do
+    # do stuff
```
Maybe we should translate this line.

nscyclone commented on ru/lessons/libraries/guardian.md:
````diff
@@ -0,0 +1,249 @@
+end
+```
+
+Реализация логина через браузерную сессию самостоятельно вызывает `encode_and_sign` внутри имплементации `Guardian.Plug.sign_in`. Потому оба приведёных примера эквивалентны.
````
`приведёных` → `приведённых`

brain-geek commented:
Thanks for the feedback, @nscyclone !

I've finally got to this translation and fixed all the issues + proofread once again and fixed some more mistakes.

nscyclone commented on ru/lessons/libraries/guardian.md:
```diff
@@ -0,0 +1,249 @@
+lang: ru
+---
+
+[Guardian](https://github.com/ueberauth/guardian) &mdash; часто используемая библиотека, основанная на [JWT](https://jwt.io/) (Javascript Web Token).
```
`Javascript` → `JSON` (#916)

brain-geek commented:
Thanks for all the feedback, @nscyclone !

Merging this one.

doomspork commented:
Thank you @brain-geek and @nscyclone!  You guys are an awesome duo 😀
